app/src/main/java/com/bignerdranch/android/androidacademy/MainActivity.kt

```diff
@@ -1,55 +1,27 @@
 package com.bignerdranch.android.androidacademy
 
-import android.content.Intent
 import android.os.Bundle
-import com.google.android.material.floatingactionbutton.FloatingActionButton
-import com.google.android.material.snackbar.Snackbar
 import androidx.appcompat.app.AppCompatActivity
-import android.view.Menu
-import android.view.MenuItem
-import kotlinx.android.synthetic.main.content_main.*
 
-class MainActivity : AppCompatActivity() {
+class MainActivity : AppCompatActivity(), OnMovieItemClickListener {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
-        setSupportActionBar(findViewById(R.id.toolbar))
-
-        findViewById<FloatingActionButton>(R.id.fab).setOnClickListener { view ->
-            Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
-                    .setAction("Action", null).show()
-        }
-
-        buttonInt.setOnClickListener {
-            val intent = Intent(this, practice_one::class.java)
-            startActivity(intent)
-        }
-
-        buttonInt2.setOnClickListener {
-            val intent = Intent(this, practice_two::class.java)
-            startActivity(intent)
-        }
-
-        moovie_btn.setOnClickListener {
-            val intent = Intent(this, MovieDetailsActivity::class.java)
-            startActivity(intent)
-        }
     }
 
-    override fun onCreateOptionsMenu(menu: Menu): Boolean {
-        // Inflate the menu; this adds items to the action bar if it is present.
-        menuInflater.inflate(R.menu.menu_main, menu)
-        return true
+    override fun onItemClickShowDetail() {
+        supportFragmentManager.beginTransaction()
+            .replace(R.id.container, FragmentMoviesDetails())
+            .addToBackStack("FragmentMoviesDetails")
+            .commit()
     }
 
-    override fun onOptionsItemSelected(item: MenuItem): Boolean {
-        // Handle action bar item clicks here. The action bar will
-        // automatically handle clicks on the Home/Up button, so long
-        // as you specify a parent activity in AndroidManifest.xml.
-        return when (item.itemId) {
-            R.id.action_settings -> true
-            else -> super.onOptionsItemSelected(item)
-        }
+    override fun onItemClickShowList() {
```

Comment:
this is incorrect to come back you shouldn't create a new instance
just supportFragmentManager.popBackstack enough